dot/parachain/availability-store/availabilitystore.go

```diff
@@ -1,33 +1,159 @@
 // Copyright 2023 ChainSafe Systems (ON)
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package availability_store
+package availabilitystore
 
 import (
 	"context"
+	"encoding/binary"
+	"encoding/json"
 
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/internal/database"
 	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/common"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-store"))
 
+const (
+	avaliableDataPrefix = "available"
+	chunkPrefix         = "chunk"
+	metaPrefix          = "meta"
+	unfinalizedPrefix   = "unfinalized"
+	pruneByTimePrefix   = "prune_by_time"
+)
+
+// AvailabilityStoreSubsystem is the struct that holds subsystem data for the availability store
 type AvailabilityStoreSubsystem struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
-	//db            interface{} // Define the actual database type
-	//config        Config      // Define the actual config type
-	//pruningConfig PruningConfig
-	//clock         Clock
-	//metrics       Metrics
+	availabilityStore   AvailabilityStore
+	//TODO: pruningConfig PruningConfig
+	//TODO: clock         Clock
+	//TODO: metrics       Metrics
+}
+
+// AvailabilityStore is the struct that holds data for the availability store
+type AvailabilityStore struct {
+	availableTable database.Table
+	chunkTable     database.Table
+	metaTable      database.Table
+	//TODO: unfinalizedTable database.Table
+	//TODO: pruneByTimeTable database.Table
+}
+
+// NewAvailabilityStore creates a new instance of AvailabilityStore
+func NewAvailabilityStore(db database.Database) (*AvailabilityStore, error) {
+	return &AvailabilityStore{
+		availableTable: database.NewTable(db, avaliableDataPrefix),
+		chunkTable:     database.NewTable(db, chunkPrefix),
+		metaTable:      database.NewTable(db, metaPrefix),
+	}, nil
+}
+
+// loadAvailableData loads available data from the availability store
+func (as *AvailabilityStore) loadAvailableData(candidate common.Hash) (AvailableData, error) {
+	resultBytes, err := as.availableTable.Get(candidate[:])
+	if err != nil {
+		return AvailableData{}, err
+	}
+	result := AvailableData{}
+	err = json.Unmarshal(resultBytes, &result)
+	return result, err
+}
```

Comment:
```suggestion
func (as *AvailabilityStore) loadAvailableData(candidate common.Hash) (*AvailableData, error) {
	resultBytes, err := as.availableTable.Get(candidate[:])
	if err != nil {
		return nil, fmt.Errorf("getting candidate %v from available table: %w", candidate, err)
	}
	result := AvailableData{}
	err = json.Unmarshal(resultBytes, &result)
	if err != nil {
		return nil, fmt.Errorf("unmarshalling available data: %w", err)
	}
	return &result, nil
}
```

- let's wrap errors
- lets return AvailableData as pointer